unix/man/man1/we-blog.1
.\" manual page for we-blog, a command wrapper for We-Blog
.\" Copyright (C) 2011-2012 Ton Kersten
.\" Copyright (C) 2010-2011 Jaromir Hradilek
.\"
.\" Permission is granted to copy, distribute and/or modify this document
.\" under the terms of the GNU Free Documentation License, Version 1.3 or
.\" any later version published by the Free Software Foundation;  with no
.\" Invariant Sections, no Front-Cover Texts, and no Back-Cover Texts.
.\"
.\" A copy  of the license is included  as a file called FDL  in the main
.\" directory of the We-Blog source package.
.TH we-blog 1 "2012-06-09" "Version 0.8" "We-Blog Documentation"
.SH NAME
we-blog \- allows you to run We-Blog utilities with a single command
.SH SYNOPSIS
.B  we-blog
.I  command
.RI [ option ...]
.PP
.B  we-blog
.BR help | man
.RI [ command ]
.PP
.B we-blog
.BR \-h | \-v
.SH DESCRIPTION
.B we-blog
provides a common way to run We-Blog utilities with a single command.
.SH OPTIONS
.SS Command Line Options
.TP
.BR \-h ", " \-\-help
Displays usage information and exits.
.TP
.BR \-v ", " \-\-version
Displays version information and exits.
.SS Available Commands
.TP
.BR init ", " in
Runs
.BR we-blog-init ,
a utility that either creates a We-Blog repository, or recovers a
previously corrupted one.
.TP
.BR config ", " cf ", " cfg
Runs
.BR we-blog-config ,
a utility that either displays, or sets We-Blog configuration options.
.TP
.B  add
Runs
.BR we-blog-add ,
a utility that adds a blog post or a page to the We-Blog repository.
.TP
.BR edit ", " ed
Runs
.BR we-blog-edit ,
a utility that edits a blog post or a page in the We-Blog repository.
.TP
.BR remove ", " rm ", " del
Runs
.BR we-blog-remove ,
a utility that removes a blog post or a page from the We-Blog
repository.
.TP
.BR list ", " ls
Runs
.BR we-blog-list ,
a utility that lists blog posts or pages in the We-Blog repository.
.TP
.BR make ", " mk
Runs
.BR we-blog-make ,
a utility that generates a blog from the We-Blog repository.
.TP
.B  log
Runs
.BR we-blog-log ,
a utility that displays the We-Blog repository log.
.TP
.B  help
Displays usage information, either for the utility itself, or for the
selected
.IR command .
.TP
.B  man
Displays a manual page, either for the utility itself, or for the selected
.IR command .
.TP
.BR version ", " vs ", " ver
Displays version information.
.SH ENVIRONMENT
.TP
.B  EDITOR
Unless the
.B  core.editor
option is set,
.B  we-blog
tries to use system-wide settings to decide which editor to use.
.SH SEE ALSO
.BR we-blog-init (1),
.BR we-blog-config (1),
.BR we-blog-add (1),
.BR we-blog-edit (1),
.BR we-blog-remove (1),
.BR we-blog-list (1),
.BR we-blog-make (1),
.BR we-blog-log (1)
.SH BUGS
To report a bug or to send a patch, please, add a new issue to the bug
tracker at <http://code.google.com/p/we-blog/issues/>, or visit the
discussion group at <http://groups.google.com/group/we-blog/>.
.SH COPYRIGHT
Copyright (C) 2011-2012 Ton Kersten
Copyright (C) 2009-2011 Jaromir Hradilek
.PP
This program is free software; see the source for copying conditions. It is
distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.


ChangeLog
2012-07-09  Ton Kersten <we-blog@tonkersten.com>
  * Changed a lot of things, but they all boil down to
    bugfixes and code reshuffeling

2011-10-07  Ton Kersten <we-blog@tonkersten.com>
  * Added 'alt' text for pages to show on mouse-over.

2011-09-21  Ton Kersten <we-blog@tonkersten.com>
  * And switched to branch 0.8 for further development

2011-09-21  Ton Kersten <we-blog@tonkersten.com>
  * Increased we-blog to version 0.7
  * Added 'smiley' converter
  * Published the project on GitHub

2011-08-15  Ton Kersten <we-blog@tonkersten.com>
  * Forked BlazeBlogger 1.2.0 to we-blog 0.5
    This because development of Blaze seems to be stalled

2011-02-18  Jaromir Hradilek <jhradilek@gmail.com>

  * INSTALL, Makefile, README, src/we-blog-add.pl, src/we-blog-config.pl,
  src/we-blog-edit.pl, src/we-blog-init.pl, src/we-blog-list.pl, src/we-blog-log.pl,
  src/we-blog-make.pl, src/we-blog-remove.pl, unix/we-blog.sh, unix/man/man1/we-blog.1,
  windows/we-blog.bat: Increased the version number.

  * unix/man/man1/we-blog.1: Updated the manual page for the command wrapper.

# $Id: ChangeLog 3 2011-10-07 13:39:05 tonk $


rpm-build/we-blog-0.9/unix/we-blog.sh
#!/bin/sh
# vi: set sw=4 ts=4 ai:
# $Id: we-blog.sh 3 2012-07-09 12:55:25 tonk $

# we-blog, a command wrapper for We-Blog
# Copyright (c) 2011-2013 Ton Kersten
# Copyright (c) 2009-2011 Jaromir Hradilek

# This program is  free software:  you can redistribute it and/or modify it
# under  the terms  of the  GNU General Public License  as published by the
# Free Software Foundation, version 3 of the License.
#
# This program  is  distributed  in the hope  that it will  be useful,  but
# WITHOUT  ANY WARRANTY;  without  even the implied  warranty of MERCHANTA-
# BILITY  or  FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
# License for more details.
#
# You should have received a copy of the  GNU General Public License  along
# with this program. If not, see <http://www.gnu.org/licenses/>.

# General script information:
NAME=${0##*/}
VERSION='0.9'

# Get the command if any, and shift command line options:
COMMAND=$1
shift

# Substitute aliases:
case "$COMMAND" in
	"ed")         COMMAND="edit";;
	"in")         COMMAND="init";;
	"ls")         COMMAND="list";;
	"mk")         COMMAND="make";;
	"rm" | "del") COMMAND="remove";;
	"cf" | "cfg") COMMAND="config";;
	"vs" | "ver") COMMAND="version";;
esac

# Parse the command and perform an appropriate action:
case "$COMMAND" in
	"add" | "log" | "edit" | "init" | "list" | "make" | "config" | "remove")
		# Run the selected utility:
		exec we-blog-$COMMAND "$@"
	;;
	"-h" | "--help" | "help")
		# Get the command, if any:
		COMMAND=$1

		# Parse the command, and display its usage:
		case "$COMMAND" in
			"add" | "log" | "edit" | "init" | "list" | "make" | "config" | "remove")
				# Display the utility usage information:
				exec we-blog-$COMMAND --help
			;;

			*)
				# Display the list of available commands:
				echo "Usage: $NAME COMMAND [OPTION...]"
				echo
				echo "Basic commands:"
				echo "  init     create or recover a We-Blog repository"
				echo "  config   display or set We-Blog configuration options"
				echo "  add      add a blog post or page to a We-Blog repository"
				echo "  edit     edit a blog post or page in a We-Blog repository"
				echo "  remove   remove a blog post or page from a We-Blog repository"
				echo "  list     list blog posts or pages in a We-Blog repository"
				echo "  make     generate a blog from a We-Blog repository"
				echo "  log      display a We-Blog repository log"
				echo
				echo "Additional commands:"
				echo "  help [COMMAND]  display usage information on the selected command"
				echo "  man [COMMAND]   display a manual page for the selected command"
				echo "  version         display version information"
				echo
				echo "Command aliases:"
				echo "  in       init"
				echo "  ed       edit"
				echo "  ls       list"
				echo "  mk       make"
				echo "  cf, cfg  config"
				echo "  rm, del  remove"
				echo "  vs, ver  version"

				# Return success:
				exit 0
			;;
		esac
	;;
	"man")
		# Get the command, if any:
		COMMAND=$1

		# Parse the command, and display its manual page:
		case "$COMMAND" in
			"add" | "log" | "edit" | "init" | "list" | "make" | "config" | "remove")
				# Display the utility usage information:
				exec man we-blog-$COMMAND
			;;
			*)
				# Display a general manual page:
				exec man we-blog
			;;
		esac
	;;
	"-v" | "--version" | "version")
		# Display version information:
		cat <<- @EOF
			We-Blog $VERSION

			Copyright (c) 2011-2013 Ton Kersten
			Copyright (c) 2009-2011 Jaromir Hradilek

			This program is free software; see the source for copying conditions. It is
			distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
			without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PAR-
			TICULAR PURPOSE.
		@EOF

		# Return success:
		exit 0
	;;
	*)
		# Respond to a wrong or missing command:
		echo "Usage: $NAME COMMAND [OPTION...]" >&2
		echo "Try \`$NAME help' for more information." >&2

		# Return failure:
		exit 22
	;;
esac
